### DISCUSSION TITLE
How to bulk import users into Users table?

### DISCUSSION THREAD

**Discussion by jdmoyer82** [Budibase/budibase#1416] [Help]
I can create a new table from importing a CSV file but I don't see a way to bulk import users into the Users table, since I can't delete the users table and create a new one from CSV.

How can I import users in bulk?

**Comment by shogunpurple** [comment]
Hi @jdmoyer82 - you cannot import users from a CSV at the moment. In the next version of budibase coming over the next few months, this will be possible through a variety of methods, but right now it cannot be done from CSV.

You can create users through an automation however, so writing a script that you trigger with an HTTP request would allow you to iterate through your CSV and create users. I know this isn't the simplest approach, but it could be a workaround.

**Reply by adamjehu** [reply]
Hi, is there any time-frame for when we may be able to bulk load users using a CSV or perhaps a list of emails using a delimiter?

**Reply by mjashanks** [reply]
@adamjehu 
It's not on our roadmap, although you could script this and use our API. The API is undocumented at the moment.

So, what you would need to do it

1. Go to the `Users` section, open dev tools in your browser, and go to the `Network` tab
2. Create a new user
3. Look in your network tab for the request `POST /api/global/users/invite`
4. copy that request, including the `cookie` and `body` 
5. Write a script in your programming language of choice, that iterates through your CSV, and runs this request for every email. You will need to modify the body, to send the current email in the iteration.

**Comment by jdmoyer82** [comment]
Thanks. I’m not a developer at all. Mind sharing more detail on how I can import users in bulk through automation? I don't know how to write scripts so is there any other way for me to do it?

**Reply by shogunpurple** [reply]
Hi Jeff, possibly a no-code automation tool like zapier, integromat or n8n would be a good approach to do this. You could give it your CSV, and have it execute HTTP requests against budibase for each row. I still think a degree of technical knowledge would be required to perform this action with these tools, however.
